confluent_kafka/avro/cached_schema_registry_client.py

```diff
@@ -42,17 +44,46 @@ class CachedSchemaRegistryClient(object):
 
     See http://confluent.io/docs/current/schema-registry/docs/intro.html
 
+    .. deprecated::
+    Use CachedSchemaRegistryClient(dict: config) instead.
+    Existing params ca_location, cert_location and key_location will be replaced with their librdkafka equivalents:
+    `ssl.ca.location`, `ssl.certificate.location` and `ssl.key.location` respectively.
+
     Errors communicating to the server will result in a ClientError being raised.
 
-    @:param: url: url to schema registry
+    :param: str|dict url: url(deprecated) to schema registry or dictionary containing client configuration
+    :param: str ca_location: File or directory path to CA certificate(s) for verifying the Schema Registry key
+    :param: str cert_location: Path to client's public key used for authentication.
+    :param: str key_location: Path to client's private key used for authentication.
+
     """
 
     def __init__(self, url, max_schemas_per_subject=1000, ca_location=None, cert_location=None, key_location=None):
-        """Construct a client by passing in the base URL of the schema registry server"""
+        # In order to maintain comparability the url(conf in future versions) param has been preserved for now.
+        conf = url
+        if isinstance(url, str):
+            conf = {
+                'url': url,
+                'ssl.ca.location': ca_location,
+                'ssl.certificate.location': cert_location,
+                'ssl.key.location': key_location
+            }
+            warnings.simplefilter('always', DeprecationWarning)  # Deprecation warnings are suppressed by default
```

Comment:
Are you really supposed to change the warnings filter from the source of warning, rather than the app?
If so; how would an app silence these?